custom_components/spotcast/spotcast_controller.py

```diff
@@ -98,24 +98,38 @@ def startSpotifyController(self, access_token: str, expires: int) -> None:
 
         self.spotifyController = sp
 
-    def getSpotifyDeviceId(self, devices_available: dict) -> None:
-        # Look for device to make sure we can start playback
-        _LOGGER.debug(
-            "devices_available: %s %s", devices_available, self.spotifyController.device
-        )
-        if devices := devices_available["devices"]:
-            for device in devices:
-                if device["id"] == self.spotifyController.device:
-                    return device["id"]
-
-        _LOGGER.error(
-            'No device with id "{}" known by Spotify'.format(
-                self.spotifyController.device
-            )
-        )
-        _LOGGER.error("Known devices: {}".format(devices))
-
-        raise HomeAssistantError("Failed to get device id from Spotify")
+    def get_device_id_from_spotify(self, hass: ha_core.HomeAssistant, spotify_user_id: str) -> dict:
+        """
+        Get the Spotify device needed to start playback on the chromecast device.
+        """
+        spotify_media_player = get_spotify_media_player(hass, spotify_user_id)
+
+        start_time = time.time()
+        timeout_s = 30
+        spotify_device = None
+        devices_available = None
+        _LOGGER.debug(f"Searching for playback device { self.spotifyController.device } on spotify")
+        # Sometimes the chromecast device might not yet be known to spotify e.g. if the chromecast is recently booted
+        # We need to wait for the chromecast to be known to spotify before we can start playback
+        max_retries = 5
+        counter = 0
+        while counter < max_retries:
+            counter += 1
+            devices_available = get_spotify_devices(spotify_media_player)
+            for spotify_device in devices_available:
+                if spotify_device["id"] == self.spotifyController.device:
+                    spotify_device = spotify_device["id"]
```

Comment:
The variable `spotify_device` of the `for` loop has the same name as the variable `spotify_device` defined in line 109.
This might cause unexpected behaviour.

Reply:
Any advance on this please?